### ISSUE TITLE
Questions about training with customdatasets

### ISSUE THREAD

**Issue by huangwan-jiayi** [CoinCheung/BiSeNet#210]
I made the following changes：
![image](https://user-images.githubusercontent.com/71338215/147355538-b4595949-acc5-4ca2-8384-71d8b19c35a8.png)
the errror:
![image](https://user-images.githubusercontent.com/71338215/147355583-e22861fa-106c-4ae5-976e-99c6e1e10dc0.png)
![image](https://user-images.githubusercontent.com/71338215/147355638-e90495fd-0b3a-4653-a6ec-5bea9d139f69.png)

![image](https://user-images.githubusercontent.com/71338215/147355601-e1bdc937-1d04-4aec-aa61-883eebb5cb5f.png)
I hope someone can help me take a look, how should I modify。Thanks！！！！！！！！！

**Comment by CoinCheung** [comment]
Hi,

Please make sure your input image sizes are divisible by 32.

**Comment by huangwan-jiayi** [comment]
> Hi,
> 
> Please make sure your input image sizes are divisible by 32.
 he worked！  thank you very much！！ If I want to train a custom crop size, where should I modify the code so that the crop ratio is not restricted by 32.thank you！！

**Comment by huangwan-jiayi** [comment]
Excuse me, but I encountered a new problem. When I started training and reported the following error, how can I solve it? Thank you.
![image](https://user-images.githubusercontent.com/71338215/147378033-e0953bc9-256a-4182-9eba-bee99be86880.png)

**Comment by CoinCheung** [comment]
You can change crop size in configuration file: https://github.com/CoinCheung/BiSeNet/blob/f9231b7c971413e6ebdfcd961fbea53417b18851/configs/bisenetv1_city.py#L15

**Comment by CoinCheung** [comment]
What is your dataset like? How many categories are in your dataset? What is the range of label values?

**Comment by huangwan-jiayi** [comment]
> What is your dataset like? How many categories are in your dataset? What is the range of label values?

it has 3 categories ，sorry，what about the last question mean？

**Comment by CoinCheung** [comment]
Please check here: https://github.com/CoinCheung/BiSeNet/blob/f9231b7c971413e6ebdfcd961fbea53417b18851/lib/cityscapes_cv2.py#L67
If your label values are within (0,1,2), you need to bypass these lines about label value mapping.

**Comment by huangwan-jiayi** [comment]
> Please check here:
> 
> https://github.com/CoinCheung/BiSeNet/blob/f9231b7c971413e6ebdfcd961fbea53417b18851/lib/cityscapes_cv2.py#L67
> 
> 
> If your label values are within (0,1,2), you need to bypass these lines about label value mapping.

Do you mean to modify this file？
![image](https://user-images.githubusercontent.com/71338215/147446077-b9066ff9-73e3-41f7-b703-83f9f260eeb2.png)
I made the following changes：
![image](https://user-images.githubusercontent.com/71338215/147446115-a37e271d-802f-4738-8d6a-11a76a2d5c65.png)
My label map：
![image](https://user-images.githubusercontent.com/71338215/147446269-53e17aa4-1d18-4d6d-b29d-c031b02250a3.png)
What changes should I make to solve this problem？

**Comment by CoinCheung** [comment]
Have you checked every label image to make sure every label image is within [0-2] ?

**Comment by huangwan-jiayi** [comment]
![image](https://user-images.githubusercontent.com/71338215/147522029-2cfcdb23-ff28-4e3a-be18-78f73afcaf68.png)
There are only two categories of label images and three categories of label images in the data set

**Comment by CoinCheung** [comment]
I mean, have you checked one by one, and make sure each picuture's value are within [0-2]? Better try with `cv2.imread(pth, 0)`:
https://github.com/CoinCheung/BiSeNet/blob/f9231b7c971413e6ebdfcd961fbea53417b18851/lib/base_dataset.py#L53

**Comment by huangwan-jiayi** [comment]
you mean this？
The output of each label map in my data set is like the following figure。

![image](https://user-images.githubusercontent.com/71338215/147543412-efb1cb8f-3af1-4365-9cd7-6069aebc4e88.png)
![image](https://user-images.githubusercontent.com/71338215/147543443-38850417-ab4c-454c-9719-dba3707d11a7.png)
I still haven't solved this problem。
Process finished with exit code 134 (interrupted by signal 6: SIGABRT)

**Comment by CoinCheung** [comment]
These are pytorch native operators, which is unlikely to have memory problem.
What is the batch size did you use to train your model？

**Comment by huangwan-jiayi** [comment]
batchsize is 8
![image](https://user-images.githubusercontent.com/71338215/147560874-b739ead8-bf4d-49ed-8bf7-c3e3be32d128.png)

**Comment by Thatboy7** [comment]
> batchsize is 8 ![image](https://user-images.githubusercontent.com/71338215/147560874-b739ead8-bf4d-49ed-8bf7-c3e3be32d128.png)

excuse me ! have you ever solved this problem?

**Comment by Thatboy7** [comment]
> batchsize is 8 ![image](https://user-images.githubusercontent.com/71338215/147560874-b739ead8-bf4d-49ed-8bf7-c3e3be32d128.png)

i meet the same question

**Comment by miscedence12** [comment]
@Thatboy7 you should guarantee your labels have one channel.
